### PR TITLE
Add support for automatically configuring AppDynamics

### PR DESCRIPTION
**Pull request by androa**
Uses the APP_NAME environment variable for configuring the AppDynamics agent.

### CODE REVIEW

File: java-10/appdynamics/README.md [hunk 1]
```diff
@@ -6,11 +6,43 @@ Basic Usage
 
 Se README for Java 10 for general usage.
 
+## Configuration of AppDynamics
+
+AppDynamics will automatically report itself by using the value of the
+environment variable `APP_NAME` as the Application name.
```
Comment (davidsteinsland) [inline java-10/appdynamics/README.md]: `APPD` or `APP`?

File: java-11/appdynamics/README.md [hunk 2]
```diff
@@ -6,11 +6,43 @@ Basic Usage
 
 Se README for Java 11 for general usage.
 
+## Configuration of AppDynamics
+
+AppDynamics will automatically report itself by using the value of the
+environment variable `APP_NAME` as the Application name.
```
Comment (davidsteinsland) [inline java-11/appdynamics/README.md]: `APPD` or `APP`?

File: java-8/appdynamics/README.md [hunk 3]
```diff
@@ -6,11 +6,43 @@ Basic Usage
 
 Se README for Java 8 for general usage.
 
+## Configuration of AppDynamics
+
+AppDynamics will automatically report itself by using the value of the
+environment variable `APP_NAME` as the Application name.
```
Comment (davidsteinsland) [inline java-8/appdynamics/README.md]: `APPD` or `APP`?

**Review by davidsteinsland — APPROVED** [review]

File: java-11/appdynamics/README.md (hunk 2 above)
Comment (androa) [reply java-11/appdynamics/README.md]: `APP_NAME` is provided today. It's a wee bit unfortunate if is removed in naiserator. Then it *can* simply be set in the Docker file though. 

But I think it is a very good practice to separate build from instance-name, and is something that should be injected at deploy time, and not at build time.

**Pull request merged**